AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -5468,6 +5470,313 @@ function New-AzDnsForwarder {
+         [bool]$StopOnAADUserLookupFailure = $true,
+
+         [Parameter(Mandatory=$false, Position=6, HelpMessage="If true, permissions will be mapped to closest available on AAD")]
+         [bool]$AutoFitSharePermissionsOnAAD = $true
+        )
+
+    # Certain accounts in a domain server will not be represented in Azure AD.
+    [String[]]$wellKnowAccountName = 'Everyone', 'BUILTIN\Administrators', 'Domain', 'Authenticated Users', 'Users', 'SYSTEM', 'Domain Admins', 'Domain Users'
+    $wellKnownAccountNamesSet = [System.Collections.Generic.HashSet[String]]::new([System.Collections.Generic.IEnumerable[String]]$wellKnowAccountName)
+
+    $roleAssignmentsDoneList = New-Object System.Collections.Generic.List[Microsoft.Azure.Commands.Resources.Models.Authorization.PSRoleAssignment]
+    $roleAssignmentsSkippedAccountsForMissingRoles = New-Object System.Collections.Generic.List[CimInstance]
+    $roleAssignmentsSkippedAccountsForMissingIdentity = New-Object System.Collections.Generic.List[CimInstance]
+    $roleAssignmentsSkippedAccountsForHavingRoleAlready = New-Object System.Collections.Generic.List[CimInstance]
+    $roleAssignmentsDoneAccounts = New-Object System.Collections.Generic.List[CimInstance]
+    $roleAssignmentsPossibleWithoutAnySkips = $True
+
+    $StorageAccountObj = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+
+    Write-Verbose -Message "Getting the local SMB share access details"
+    $localSmbShareAccess = Get-SmbShareAccess -Name $LocalShareName
+
+    if ($localSmbShareAccess -eq $null)
+    {
+        Write-Error -Message "Could not find share with name $LocalShareName." -ErrorAction stop
+    }
+
+    Write-Host "Local SMB share access details"
+
+    $localSmbShareAccess | Format-Table | Out-String|% {Write-Host $_}
+
+    # Run through ACL of the local share.
+    foreach($smbShareAccessControl in $localSmbShareAccess)
+    {
+        $account=$smbShareAccessControl.AccountName
+        $strAccessRight =[string] $smbShareAccessControl.AccessRight
+        $strAccessControlType = [string] $smbShareAccessControl.AccessControlType
+        
+        if($wellKnownAccountNamesSet.Contains($account))
+        {
+            $roleAssignmentsSkippedAccountsForMissingIdentity.Add($smbShareAccessControl)
+            continue
+        }
+
+        $objUser = New-Object System.Security.Principal.NTAccount($account)
+        $strSID = $objUser.Translate([System.Security.Principal.SecurityIdentifier])
+
+        Write-Verbose -Message "Mapping domain user/group - $account to its corresponding identity on Azure AAD"
+
+        #Geting the OID of domain user/group using its SID
+        try
+        {
+            $aadUser = Get-AzureADUser -Filter "OnPremisesSecurityIdentifier eq '$strSID'"
+        }
+        catch
+        {
+            throw

```

Comment:
Is the try-catch needed if the error is always thrown out?

Reply:
my bad. Missed this